integrationtests/tests/fixtures/mod.rs

```diff
@@ -1020,60 +1038,59 @@ impl FederationTest {
         Ok(())
     }
 
+    /// Does any of the modules return consensus proposal that forces a new epoch
     #[allow(clippy::await_holding_refcell_ref)]
     pub async fn has_pending_epoch(&self) -> bool {
         for server in &self.servers {
-            if server
+            let mut server = server.borrow_mut();
+            let fedimint_server = &mut server.fedimint;
+
+            // Pending transactions will trigger an epoch
+            if let Ok(tx) = fedimint_server.tx_receiver.get_mut().as_mut().try_recv() {
+                fedimint_server
+                    .consensus
+                    .tx_sender
+                    .send(tx)
+                    .await
+                    .expect("Can send");
+                return true;
+            }
```

Comment:
`fedimint_server.tx_receiver` is a `Peekable` to avoid exactly this hack. You should be able to avoid re-sending and potentially re-ordering transactions this way:

```rust
if let Some(Some(_)) fedimint_server.tx_receiver.peek().now_or_never() {
    return true;
}
```

Reply:
Done.